
Due to 2.6 kernel headers we have to mess a bit with the older XFree86
linux support files ... :-(

  - Rene Rebe <rene@exactcode.de>

--- ./xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c.orig	2004-11-07 19:11:14.037468118 +0000
+++ ./xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c	2004-11-07 19:12:40.379726573 +0000
@@ -73,11 +73,7 @@
 #endif
 
 /* Deal with spurious kernel header change */
-#if defined(LINUX_VERSION_CODE) && defined(KERNEL_VERSION)
-# if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,42)
-#  define rate period
-# endif
-#endif
+#define rate period
 
 static int
 KDKBDREP_ioctl_ok(int rate, int delay) {
--- ./xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c.orig	2004-11-07 19:13:59.942061811 +0000
+++ ./xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c	2004-11-07 19:19:09.523762656 +0000
@@ -107,7 +107,7 @@
    struct kbd_repeat kbdrep_s;
 
    /* don't change, just test */
-   kbdrep_s.rate = -1;
+   kbdrep_s.period = -1;
    kbdrep_s.delay = -1;
    if (ioctl( 0, KDKBDREP, &kbdrep_s )) {
        return 0;
@@ -115,11 +115,11 @@
 
    /* do the change */
    if (rate == 0)				/* switch repeat off */
-     kbdrep_s.rate = 0;
+     kbdrep_s.period = 0;
    else
-     kbdrep_s.rate  = 10000 / rate;		/* convert cps to msec */
-   if (kbdrep_s.rate < 1)
-     kbdrep_s.rate = 1;
+     kbdrep_s.period  = 10000 / rate;		/* convert cps to msec */
+   if (kbdrep_s.period < 1)
+     kbdrep_s.period = 1;
    kbdrep_s.delay = delay;
    if (kbdrep_s.delay < 1)
      kbdrep_s.delay = 1;
